src/mon/OSDMonitor.cc

```diff
@@ -1625,18 +1625,21 @@ bool OSDMonitor::can_mark_in(int i)
   return true;
 }
 
-void OSDMonitor::check_failures(utime_t now)
+bool OSDMonitor::check_failures(utime_t now)
 {
+  bool propose_now = false;
   for (map<int,failure_info_t>::iterator p = failure_info.begin();
        p != failure_info.end();
        ++p) {
     if (can_mark_down(p->first)) {
-      check_failure(now, p->first, p->second);
+      check_failure(now, p->first, p->second, propose_now ? NULL : &propose_now);
     }
   }
+  return propose_now;
```

Comment:
check_failure() may also return true in this case: https://github.com/ceph/ceph/pull/7942/files#diff-0a5db46a44ae9900e226289a810f10e8R1705

In the above case we may don't want to trigger a duplicated propose, I guess.